kernel/src/process.rs

```diff
@@ -1804,8 +1847,9 @@ impl<C: 'static + Chip> Process<'a, C> {
                 process
                     .current_stack_pointer
                     .set(new_stack_pointer as *mut u8);
-                process.debug_set_max_stack_depth();
-                process.stored_state.set(stored_state);
```

Comment:
I think this line needs to stay (though this will be moot when your other PR is merged).

Reply:
Yes, maybe I should have included the commit in #1826 in this PR too.

Reply:
Done.